### PR TITLE
TransactionScheduler: Clean up dead code

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
Some dead_code markers that are no longer necessary.
Some code that is not necessary.

#### Summary of Changes
- Remove dead_code markers for code that is used
- Remove forward worker code we ended up not using
- Add dead_code marker on batch cus - want to use this, but need the earlier meta for this to be even somewhat reliable and useful

Fixes #

### CODE REVIEW

File: core/src/banking_stage/forward_worker.rs
Comment (taozhu-chicago) [inline core/src/banking_stage/forward_worker.rs:1]: I'm sure it'd greatly simply scheduler by removing forwarding logic; can you remind me why we feel confident to move on without forwarder?

**Review by taozhu-chicago — COMMENTED** [review]
looks good to remove unused code. Just want to confirm it is OK to remove forwarding logic.

File: core/src/banking_stage/forward_worker.rs
Comment (apfitzge) [reply core/src/banking_stage/forward_worker.rs:1]: forwarding logic is already pulled out.
RPCs just forward to next N leaders anyway, >90% of forwarded packets are duplicate on test and mainnet.

Comment (taozhu-chicago) [reply core/src/banking_stage/forward_worker.rs:1]: Ah right, I recall we talked about that. Are community or team in large aware of new scheduler does not do forwarding, but rely on RPCs to do so? This change of new scheduler doesn't impact consensus directly, but could still impact cluster liveness if forwarding not done correctly. Probably it has been discussed/communicated already, but if not, probably worth doing so?

Comment (apfitzge) [reply core/src/banking_stage/forward_worker.rs:1]: Yeah probably need to communicate that more clearly. Will try to find a place on discord to do so properly.

**Pull request closed without merging**